Struct update does not work with dereferencing

<!-- Please make sure to run `v up` before reporting any issues as it may have already been fixed.
     It's also advisable to update all relevant modules using `v outdated` and `v install` -->

<!-- You can use `v doctor` to fill up the next fields -->
**V version: 0.3.0 72056f3** 
**OS: ArchLinux** 

<!-- Please include the standalone minimal reproducible code.
If not, it may be closed by the administrators. -->
**What did you do?**
```v
struct Foo{
	s string
	n int
}

fn main(){
	f := &Foo{s:'AA'}
	b := Foo{
		...*f
		n: 3
	}
}
```

**What did you expect to see?**

No error
 
**What did you see instead?**
```
==================
/usr/lib64/crt1.o: error: Invalid relocation entry [15] '.rela.debug_info' @ 00000504
tcc: error: file 'crt1.o' not found
/tmp/v_1000/bug.7482888345384196006.tmp.c:12075: error: struct or union expected
...
==================
```

--- a/vlib/v/gen/c/struct.v
+++ b/vlib/v/gen/c/struct.v
@@ -195,7 +195,9 @@ fn (mut g Gen) struct_init(node ast.StructInit) {
 				if is_update_tmp_var {
 					g.write(tmp_update_var)
 				} else {
+					g.write('(')
 					g.expr(node.update_expr)
+					g.write(')')
 				}
 				if node.update_expr_type.is_ptr() {
 					g.write('->')

--- a/vlib/v/tests/struct_init_with_update_test.v
+++ b/vlib/v/tests/struct_init_with_update_test.v
@@ -31,13 +31,15 @@ fn test_struct_init_with_update_expr() {
 	assert o.age == 21
 }
 
-struct Foo{
+struct Foo {
 	s string
 	n int
 }
 
-fn test_struct_init_with_update_expr2(){
-	f := &Foo{s:'AA'}
+fn test_struct_init_with_update_expr2() {
+	f := &Foo{
+		s: 'AA'
+	}
 	b := Foo{
 		...*f
 		n: 3

--- a/vlib/v/tests/struct_init_with_update_test.v
+++ b/vlib/v/tests/struct_init_with_update_test.v
@@ -30,3 +30,19 @@ fn test_struct_init_with_update_expr() {
 	assert o.height == 175
 	assert o.age == 21
 }
+
+struct Foo{
+	s string
+	n int
+}
+
+fn test_struct_init_with_update_expr2(){
+	f := &Foo{s:'AA'}
+	b := Foo{
+		...*f
+		n: 3
+	}
+	println(b)
+	assert b.s == 'AA'
+	assert b.n == 3
+}

